flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/nodes/dataset/DataSetAggregate.scala

```diff
@@ -157,4 +161,41 @@ class DataSetAggregate(
       case _ => result
     }
   }
+
+  /**
+    * Dummy [[Row]] into a [[DataSet]] for result after map operations.
+    * @param mapOperator after which insert dummy records
+    * @param tableEnv [[BatchTableEnvironment]] for getting rel builder and type factory
+    * @tparam IN mapOperator input type
+    * @tparam OUT mapOperator output type
+    * @return DataSet of type Row is contains null literals for columns
+    */
+  private def dummyRow[IN,OUT](
+      mapOperator: MapOperator[IN,OUT],
+      tableEnv: BatchTableEnvironment): DataSet[Row] = {
+
+    val builder: RelDataTypeFactory.FieldInfoBuilder = getCluster.getTypeFactory.builder
+    val rowInfo = mapOperator.getResultType.asInstanceOf[RowTypeInfo]
+
+    val nullLiterals :ImmutableList[ImmutableList[RexLiteral]] =
+      ImmutableList.of(ImmutableList.copyOf[RexLiteral](
+        for (fieldName <- rowInfo.getFieldNames)
+          yield {
+            val columnType = tableEnv.getTypeFactory
+              .createTypeFromTypeInfo(rowInfo.getTypeAt(fieldName))
+            builder.add(fieldName, columnType)
+            tableEnv.getRelBuilder.getRexBuilder
+              .makeLiteral(null,columnType,false).asInstanceOf[RexLiteral]
+          }))
+
+    val dataType = builder.build()
+
+    val relNode = RelFactories.DEFAULT_VALUES_FACTORY
+      .createValues(getCluster, dataType, nullLiterals)
+
+    DataSetValuesRule.INSTANCE.asInstanceOf[DataSetValuesRule]
```

Comment:
I proposed to implement a `RelOptRule` that replaces
```
LogicalAggregete(groupingsets=null) <- X
```
by 
```
                                                        /-- X
LogialAggregtate(groupingsets=null) <-- LogicalUnion <-<
                                                        \-- LogialValues(nullRow)
```

If we have such a rule, `LogicalValues` will be automatically handled by `DataSetValues` and we do not need to add any code to generate DataSet code. The only thing we need to add is one more condition to `DataSetAggregateRule.match()` to prevent a translation of a global aggregate without union-values(null) input.

If we use a rule for the translation, we can also implement the tests via `TableTestBase` because the `LogicalValues(null)` appears in the plan.